src/shared/configUtils.ts

```diff
@@ -35,27 +48,29 @@ export class ConfigUtils {
     return this.#globalConfig;
   }
 
-  public static async getOrCreateIdentityToken(): Promise<string> {
-    let token = await this.getIdentityToken();
-    if (!token) {
-      token = CryptoUtils.generateIdentityToken();
-      await this.writeIdentityToken(token);
+  public static async getOrCreateIdentityToken(username: string, connection: Connection): Promise<string> {
+    let identityData = await this.getIdentityData();
+    if (!identityData) {
+      const token = CryptoUtils.generateIdentityToken();
+      const entityId = await this.saveIdentityTokenToServer(token, connection);
+      identityData = new LocalWebServerIdentityData(token);
+      identityData.usernameToServerEntityIdMap[username] = entityId;
```

Comment:
If you update the constructor as above, then you can just have `identityData = new LocalWebServerIdentityData(token, { username: entityId });`